File: python/test/test_scan.py
import ThrustRTC as trtc

darr = trtc.device_vector_from_list([1, 0, 2, 2, 1, 3], 'int32_t')
trtc.Inclusive_Scan(darr, darr)
print (darr.to_host())

darr = trtc.device_vector_from_list([-5, 0, 2, -3, 2, 4, 0, -1, 2, 8], 'int32_t')
trtc.Inclusive_Scan(darr, darr, trtc.Maximum())
print (darr.to_host())

darr = trtc.device_vector_from_list([1, 0, 2, 2, 1, 3], 'int32_t')
trtc.Exclusive_Scan(darr, darr)
print (darr.to_host())

darr = trtc.device_vector_from_list([1, 0, 2, 2, 1, 3], 'int32_t')
trtc.Exclusive_Scan(darr, darr, trtc.DVInt32(4))
print (darr.to_host())

darr = trtc.device_vector_from_list([-5, 0, 2, -3, 2, 4, 0, -1, 2, 8], 'int32_t')
trtc.Exclusive_Scan(darr, darr, trtc.DVInt32(1), trtc.Maximum())
print (darr.to_host())

File: java/test/test_scan.java
import java.util.*;
import JThrustRTC.*;

public class test_scan
{
	public static void main(String[] args) 
	{
		{
			DVVector darr = new DVVector(new int[] { 1, 0, 2, 2, 1, 3 });
			TRTC.Inclusive_Scan(darr, darr);
			System.out.println(Arrays.toString((int[])darr.to_host()));
		}

		{
			DVVector darr = new DVVector(new int[] { -5, 0, 2, -3, 2, 4, 0, -1, 2, 8 });
			TRTC.Inclusive_Scan(darr, darr, new Functor("Maximum"));
			System.out.println(Arrays.toString((int[])darr.to_host()));
		}

		{
			DVVector darr = new DVVector(new int[] { 1, 0, 2, 2, 1, 3 });
			TRTC.Exclusive_Scan(darr, darr);
			System.out.println(Arrays.toString((int[])darr.to_host()));
		}

		{
			DVVector darr = new DVVector(new int[] { 1, 0, 2, 2, 1, 3 });
			TRTC.Exclusive_Scan(darr, darr, new DVInt32(4));
			System.out.println(Arrays.toString((int[])darr.to_host()));
		}

		{
			DVVector darr = new DVVector(new int[] { -5, 0, 2, -3, 2, 4, 0, -1, 2, 8 });
			TRTC.Exclusive_Scan(darr, darr, new DVInt32(1), new Functor("Maximum"));
			System.out.println(Arrays.toString((int[])darr.to_host()));
		}

	}
}
